apps/mull-ui/src/app/pages/waste-recognition-page/waste-recognition-page.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { toast } from 'react-toastify';
+import { MULL_MODEL_URL } from '../../../constants';
+import { drawDetectionIcons } from '../../../utilities';
+import { useToast } from '../../hooks/useToast';
+import { TensorflowJsModel } from '../../services/tfjs.model';
+import './waste-recognition-page.scss';
+
+/* eslint-disable-next-line */
+export interface WasteRecognitionPageProps {}
+
+export function WasteRecognitionPage(props: WasteRecognitionPageProps) {
+  const modelRef = useRef<TensorflowJsModel>(null);
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+  const videoRef = useRef<HTMLVideoElement>(null);
+  const streamRef = useRef<MediaStream>(null);
+  const { notifyToast, updateToast } = useToast();
+
+  const [videoLoading, setVideoLoading] = useState<boolean>(true);
+  const [modelLoading, setModelLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    setup();
+    return () => {
+      modelRef.current.dispose();
+      if (streamRef.current && streamRef.current.getTracks().length > 0) {
+        streamRef.current.getTracks()[0].stop();
+      }
+    };
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, []);
```

Comment:
```suggestion
  });
```

Reply:
The problem with this is that I only want the code I wrote in this useeffect to run once, not every render (which breaks my code, since I only expect to be loading the model in the useeffect and then disposing of it in the returned callback once). The linting was complaining that I wasn't adding `setup` to the dependencies, which is why I disabled that linting error.

If I do add `setup` to the dependencies list as the linting suggests, it just ends up running every render, since the `setup` function is redefined on every render.

Reply:
Right makes sense